[ENH] BEP001 - qMRI maps and some additional metadata (#690)
bids-standard/bids-specification: src/99-appendices/11-qmri.md

| TB1EPI                  | `EchoTime`, `FlipAngle`, `TotalReadoutTime`, `MixingTime`                                            |
| TB1AFI                  | `RepetitionTime`                                                                                     |
| TB1TFL                  |                                                                                                      |
| TB1RMF                  |                                                                                                      |
| TB1SRGE<sup>*</sup>     | `FlipAngle`, `InversionTime`, `RepetitionTimeExcitation`, `RepetitionTimePreperation`, `NumberShots` |
| RB1COR                  |                                                                                                      |

<sup>*</sup> Please see TB1SRGE-specific notes for the calculation of `NumberShots`.

### Meta-data requirements for qMRI maps

As qMRI maps are stored as derivatives, they are subjected to the metadata requirements of 
[derived datasets](03-modality-agnostic-files.html#derived-dataset-and-pipeline-description).

An example `dataset_description.json` for a qMRI map derivatives folder: 

> ```

Reviewer: for code blocks that have no specific syntax such as python, or json, .. please use "text"

```suggestion
```text
```

Author: https://github.com/bids-standard/bids-specification/pull/690/commits/274e1f82eade7d6d74d59bb6b5d9ebcf0b4c34a4

 ds-example/
 └── derivatives/
     └── qMRLab/
         ├── dataset_description.json
         └── sub-01/
             └── anat/
                 ├── sub-01_T1map.nii.gz
                 ├── sub-01_T1map.json
                 ├── sub-01_M0map.nii.gz
                 └── sub-01_M0map.json
```

> 
> ```
> dataset_description.json: 
> {

Reviewer: please pull the description of the example to follow out of the code block, and declare the "syntax highlighting style" of the codeblock (JSON in this case)

```suggestion
dataset_description.json: 

```json
{
```

Author: https://github.com/bids-standard/bids-specification/pull/690/commits/274e1f82eade7d6d74d59bb6b5d9ebcf0b4c34a4

values in a `T1map` become elusive.

* All the acquisition parameters that are constant across the files in a file collection
are RECOMMENDED to be added to the sidecar json of the qMRI maps.
* Relevant acquisition parameters that vary across files in a qMRI file collection are
RECOMMENDED to be added to the sidecar json of the qMRI map **`in array form`**.
* The JSON file accompanying a qMRI map which is obtained by using open-source software
is RECOMMENDED to include additional metadata fields listed in the following table:

| **Field name**                  | **Definition**                                                                                                               |
| ------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- |
| `BasedOn`                       | List of files in a file collection to generate the map. Fieldmaps are also listed, if involved in the processing.            |
| `EstimationReference`           | Reference to the study/studies on which the implementation is based.                                                         |

> | `EstimationAlgorithm`           | Type of algoritm used to perform fitting (e.g. linear, non-linear, LM etc.)                                                  |

Reviewer: please revise to not use latin anywhere. To make the specification more accessible, we try to replace e.g., with "for example" (or similar), and etc. by either dropping it completely, or with for example "and so forth"

Please check out: https://the-turing-way.netlify.app/community-handbook/style.html#avoid-latin-abbreviation

Author: 94584e95d107b033e2c619924741f97c2c36b04a

| `Units`                         | Units of the maps, in accordance with the BIDS specification.                                                                |

Example:

```Text
sub-01_T1map.nii.gz
sub-01_T1map.json
```

```
sub-01_T1map.json

> {

Reviewer: same comment as above (code syntax, and pulling description out of code block)

Author: https://github.com/bids-standard/bids-specification/pull/690/commits/274e1f82eade7d6d74d59bb6b5d9ebcf0b4c34a4

<<Parameter injected by the software/pipeline>>

"BasedOn":["anat/sub-01_fa-1_VFA.nii.gz",
           "anat/sub-01_fa-2_VFA.nii.gz",
           "anat/sub-01_fa-3_VFA.nii.gz",
           "anat/sub-01_fa-4_VFA.nii.gz",
           "fmap/sub-01_TB1map.nii.gz"],
"EstimationPaper":"Deoni et. al.MRM, 2015",
"EstimationAlgorithm":"Linear",
"Units": "second",

<<Parameters that are constant across files in the (parent) file collection>>

“MagneticFieldStrength”: “3”,

> “Manufacturer”: “Siemens”,

Reviewer: pedantic comment incoming: please use `"` instead of `“`

Author: https://github.com/bids-standard/bids-specification/pull/690/commits/d16bd402834819f489317bb104383f76daac8358

<sup>*</sup> [`entity`/`fixed`] denotes whether the OPTIONAL metadata that forms a new
flavor of qMRI applicaiton for the respective suffix varies across files of a file colleciton
(which calls for using a linking entity) or fixed. If former is the case, the entity is to be
added to the files in that file collection. Note that this addition MUST be allowed by the
priority levels given for that suffix in the [`entity table`](../99-appendices/04-entity-table.md). If latter (`fixed`) is the case,
filenames will remain the same; however, the optional metadata (third column) may
define the flavor of the application (fourth column) along with the conditional value of a
required metadata field (second column).

A derived qMRI application becomes avaiable if all the optional metadata fields
listed for the respective file collection suffix are provided for the data. In addition,
conditional rules based on the value of a given required metada field can be set
for the description of a derived qMRI application. Note that the value of this
required metadata is fixed across constituent images of a file collection and defined

> in the _Method-specific priority levels for qMRI file collections_ section above.

Reviewer: please use a link to that section here, instead of italics

Author: 0630127aa5b55211a64bad4a63cca0c127ad8c67

# Appendix XI: Quantitative MRI

Quantitative MRI (qMRI) is a collection of methods aiming at generating parametric maps
that can characterize underlying tissue properties.
Unlike that of conventional MR images (for example, `T1w`, `T2w`),

> intensity values of quantitative maps are not represented in an arbitrary grayscale range.

Reviewer 2: Can you drop the term "grayscale" or define it? As I interpret it, anything without multiple axes/values per element would naturally be grayscale. If that is a bad interpretation, which is _very_ possible, then it would be good to have some context about why that is.

Author: 0776ebf

| MEGRE                | `EchoTime`                                                                                                                   |                            |
| MTR                  | `MTState`                                                                                                                    |                            |
| MTS                  | `FlipAngle`, `MTState`, `RepetitionTimeExcitation`                                                                           |                            |
| MPM                  | `FlipAngle`, `MTState`, `RepetitionTimeExcitation`                                                                           | `EchoTime`                 |

<sup>*</sup> Please see MP2RAGE-specific notes for the calculation of `NumberShots` and regarding the
organization of `UNIT1` image.

Explanation of the table:

-   The metadata fields listed in the REQUIRED column are needed to perform a minimum viable qMRI processing for the corresponding `file collection`.

-   Note that some of the metadata fields may be constant across different files in a file collection,

>     yet still required as an input (for example, `NumberShots` in `MP2AGE`).

Reviewer 2: `MP2AGE` --> `MP2RAGE`

Author: 8a327a9

>     Such metadata fields MUST be provided in the accompanying json files.

Reviewer 3: ```suggestion
    Such metadata fields MUST be provided in the accompanying JSON files.
```

-   The metadata fields listed in the OPTIONAL column can be used to form different flavors of an existing file collection suffix,
    dispensing with the need for introducing a new suffix.

>     See [deriving the intended qMRI application from an ambiguos file collection](#Deriving-the-intended-qMRI-application-from-an-ambiguous-file-collection)

Reviewer 2: ambiguos --> ambiguous

|-------------------------|------------------------------------------------------------------------------------------------------|
| TB1DAM                  | `FlipAngle`                                                                                          |
| TB1EPI                  | `EchoTime`, `FlipAngle`, `TotalReadoutTime`, `MixingTime`                                            |
| TB1AFI                  | `RepetitionTime`                                                                                     |
| TB1TFL                  |                                                                                                      |
| TB1RMF                  |                                                                                                      |
| TB1SRGE<sup>*</sup>     | `FlipAngle`, `InversionTime`, `RepetitionTimeExcitation`, `RepetitionTimePreperation`, `NumberShots` |
| RB1COR                  |                                                                                                      |

<sup>*</sup> Please see TB1SRGE-specific notes for the calculation of `NumberShots`.

> ### Meta-data requirements for qMRI maps

Reviewer 2: Meta-data --> Metadata

Author: 43edcab

| `BasedOn`                       | List of files in a file collection to generate the map. Fieldmaps are also listed, if involved in the processing.            |
| `EstimationReference`           | Reference to the study/studies on which the implementation is based.                                                         |
| `EstimationAlgorithm`           | Type of algoritm used to perform fitting (for example, linear, non-linear, LM and such)                                      |
| `Units`                         | Units of the maps, in accordance with the BIDS specification.                                                                |

Example:

```text
sub-01_T1map.nii.gz
sub-01_T1map.json
```

sub-01_T1map.json:

```text
{

<<Parameter injected by the software/pipeline>>

"BasedOn":["anat/sub-01_fa-1_VFA.nii.gz",
           "anat/sub-01_fa-2_VFA.nii.gz",
           "anat/sub-01_fa-3_VFA.nii.gz",
           "anat/sub-01_fa-4_VFA.nii.gz",
           "fmap/sub-01_TB1map.nii.gz"],
"EstimationPaper":"Deoni et. al.MRM, 2015",
"EstimationAlgorithm":"Linear",
"Units": "second",

<<Parameters that are constant across files in the (parent) file collection>>

> "MagneticFieldStrength”: "3”,

Reviewer 2: Can you replace instances of `”` with `"`?

Author: 8f3a13e

"Manufacturer”: "Siemens”,
"ManufacturerModelName”: "TrioTim”,
"InstitutionName”: "xxx”,
"PulseSequenceType”: "SPGR”,
"PulseSequenceDetails”: "Information beyond the sequence type that identifies
 specific pulse sequence used (VB version, if not standard, Siemens WIP XXX
 ersion ### sequence written by xx using a version compiled on mm/dd/yyyy/)”,
"RepetitionTimeExcitation": "35",
"EchoTime": "2.86",
"SliceThickness": "5",

<<Relevant parameters that vary across the linking entity of the (parent) file collection>>

"FlipAngle": ["5","10","15","20"]

}
```

## Deriving the intended qMRI application from an ambiguous file collection

> Certain file colleciton suffixes may refer to a generic data collection regime such as variable flip angle (VFA),

Reviewer 2: colleciton --> collection

the inheritor qMRI method is listed in the table below instead of introducing a new file collection suffix.
This approach aims at:

-   preventing the list of available suffixes from over-proliferation,
-   providing qMRI-focused BIDS applications with a set of meta-data driven rules to infer possible fitting options,
-   keeping an inheritance track of the qMRI methods described within the specification.

| **File-collection suffix** | **If REQUIRED metadata == Value** | **OPTIONAL metadata [`entity`/`fixed`]**   | **Derived application name (NOT a suffix)** |
|----------------------------|-----------------------------------|--------------------------------------------|---------------------------------------------|
| VFA                        | `PulseSequenceType` == `SPGR`     |                                            | DESPOT1                                     |
| VFA                        | `PulseSequenceType` == `SSFP`     | `SpoilingRFPhaseIncrement` [`fixed`]       | DESPOT2                                     |
| MP2RAGE                    |                                   | `EchoTime` [`echo`]                        | MP2RAGE-ME                                  |

> | MPM                        |                                   | `EchoTime` [`echo`]                        | MPM-ME                                      |

Reviewer: The style you use here to mark settings does not work well with Markdown:

- [`entity`/`fixed`]
- [`fixed`]
- and so on

because Markdown thinks you are starting a link, and it's looking for the trailing brackets `[link-target](trailing-brackets)`, that's why the CI fails for this file.

Can you think of another (markdown-compatible) way to communicate what you want?

Reviewer 4: How about using simple brackets instead?

Author: 42cb86b

Reviewer: that seems to have worked, thanks!

> <sup>*</sup> (`entity`/`fixed`) denotes whether the OPTIONAL metadata that forms a new

Reviewer 3: I'm assuming the `<sup>*</sup>` was missing in the table header because it caused formatting issues?

```suggestion
In this table, (`entity`/`fixed`) denotes whether the OPTIONAL metadata that forms a new
```

> flavor of qMRI applicaiton for the respective suffix varies across files of a file colleciton

Reviewer 3: ```suggestion
flavor of qMRI applicaiton for the respective suffix varies across files of a file collection
```

(which calls for using a linking entity) or fixed. If former is the case, the entity is to be
added to the files in that file collection. Note that this addition MUST be allowed by the
priority levels given for that suffix in the [`entity table`](./04-entity-table.md). If latter (`fixed`) is the case,
filenames will remain the same; however, the optional metadata (third column) may
define the flavor of the application (fourth column) along with the conditional value of a
required metadata field (second column).

A derived qMRI application becomes avaiable if all the optional metadata fields
listed for the respective file collection suffix are provided for the data. In addition,
conditional rules based on the value of a given required metada field can be set
for the description of a derived qMRI application. Note that the value of this
required metadata is fixed across constituent images of a file collection and defined

> in [this section](#Method-specific-priority-levels-for-qMRI-file-collections).

Reviewer 3: ```suggestion
in [Method-specific priority levels for qMRI file collections](#Method-specific-priority-levels-for-qMRI-file-collections).
```

For example, if the optional metadata field of `PulseSequenceType` is SPGR
for a collection of anatomical images listed by the `VFA` suffix, the data
qualifies for `DESPOT1` T1 fitting. For the same suffix, if the `PulseSequenceType`
metadata field has the value of `SSFP`, and the `SpoilingRFPhaseIncrement` is
provided as a metadata field, then the dataset becomes eligible for `DESPOT2`
T2 fitting application.

Please note that optional metadata fields listed in the [deriving the intended qMRI
application from an ambiguous file collection table](#Deriving-the-intended-qMRI-application-from-an-ambiguous-file-collection) are included in the optional (third)
column of [the priority levels table](#Method-specific-priority-levels-for-qMRI-file-collections) for the consistency of this appendix.

## Introducing a new qMRI file collection

> If a qMRI application can not be interpreted as subset of an already existing suffix

Reviewer 2: can not --> cannot

Reviewer 2: "as subset" --> "as a subset" (although I think "subtype" might be better, WDYT?)

Author: 35f6375 a683f40

    [deriving the intended qMRI application from an ambiguous file collection](#Deriving-the-intended-qMRI-application-from-an-ambiguous-file-collection)
    and the
    [anatomy data priority levels](#Anatomy-imaging-data)
    sections are extended instead of introducing a new suffix.

## Application-specific notes for qMRI file collections

### Anatomy imaging data

General notes:

-   Some BIDS metadata field values are calculated based on the values of other metadata fields that are not listed as required fields.
    These fields include: `NumberShots`.
    The calculation of the values may depend on the type of the acquisition.
    These acquisitions include: `MP2RAGE` and `TB1SRGE`.

#### `MP2RAGE` specific notes

##### `UNIT1` images

Although the `UNIT1` image is provided as an output by the acquisition sequence, it is used
as an input to offline calculation of a `T1map` using a dictionary lookup approach. However,
`complex` data is needed for an accurate calculation of the `UNIT1` image, which is not commonly
provided by the stock sequence. Instead, the `magnitude` and `phase` images are exported. Please

> see the relevant discussion [here](https://github.com/qMRLab/qMRLab/issues/255).

Reviewer 3: I generally recommend more informative link text.

```suggestion
see the relevant discussion at [qMRLab issue #255](https://github.com/qMRLab/qMRLab/issues/255).
```

Therefore, the `UNIT1` image provided by the scanner is RECOMMENDED to be stored under the `anat`
raw dataset directory along with the `MP2RAGE` file collection and to be used as the primary input
for quantifying a `T1map`.

If an additional `UNIT1` image is calculated offline, then the output is to be stored in the
`derivatives` folder with neccesary provenance information.

##### `NumberShots` metadata field

> Note that type of `NumberShots` field can be either a `number` or an `array of numbers`.

Reviewer 3: ```suggestion
Note that the type of `NumberShots` field can be either a `number` or an `array of numbers`.
```

> -   If a single `number` is to be provided, this should correspond to the number of `SlicesPerSlab` or `ReconMatrixPE`.

Reviewer 2: "is to be provided" --> "is provided"

Author: 10833a8

    However, in this case, `SlicePartialFourier` or `PartialFourierPE` fraction is needed
    to calculate the number of partitions `before` and `after` of the k-space center to calculate a T1 map.

-   If `before/after` calculation is performed during the BIDS conversion of the `MP2RAGE` data,
    then the value of `NumberShots` metadata field can be given as a 1X2 array,
    with first entry corresponding to `before` and the second to the `after`.

Formula:

If NumberShots is an array of numbers such that `"NumberShots": [before, after]`,
the values of `before` and `after` are calculated as follows:

```text
before = SlicesPerSlab*(SlicePartialFourier - 0.5)
after  = SlicesPerSlab/2
```

> See this [reference](https://github.com/JosePMarques/MP2RAGE-related-scripts/blob/a405df30ac2c617d29d8b1b16025aaa911e86370/func/bids_T1B1correct.m#L16).

Reviewer 3: Maybe?

```suggestion
See this [reference implementation](https://github.com/JosePMarques/MP2RAGE-related-scripts/blob/a405df30ac2c617d29d8b1b16025aaa911e86370/func/bids_T1B1correct.m#L16).
```

##### Other metadata fields

> The value of the `RepetitionTimeExcitation` is not commonly found in the DICOM files.

Reviewer 2: "value of the `RepetitionTimeExcitation`" --> "value of the `RepetitionTimeExcitation` field"

Author: 3f33e1d

When accessible, the value of `EchoSpacing` corresponds to this metadata.
When not accessible, `2 X EchoTime` can be used as a surrogate.

> Further information about other `MP2RAGE` qMRI protocol fields can be found [here](https://qmrlab.readthedocs.io/en/master/protocols.html#mp2rage).

Reviewer 3: ```suggestion
Further information about other `MP2RAGE` qMRI protocol fields can be found in the
[qMRLab documentation](https://qmrlab.readthedocs.io/en/master/protocols.html#mp2rage).
```

#### `TB1SRGE` specific notes

Calculation of `before` and `after` entries for `NumberShots` metadata field of `TB1SRGE` is more involved than that of `MP2RAGE`.

> The formula can be found [here](https://github.com/JosePMarques/MP2RAGE-related-scripts/blob/a405df30ac2c617d29d8b1b16025aaa911e86370/DemoForR1Correction.m#L17),

Reviewer 3: ```suggestion
The formula can be found in a
[reference implementation](https://github.com/JosePMarques/MP2RAGE-related-scripts/blob/a405df30ac2c617d29d8b1b16025aaa911e86370/DemoForR1Correction.m#L17),
```

which requires information about `BaseResolution` (that is, image matrix size in PE direction),
partial Fourier fraction in the PE direction, number of reference lines for parallel imaging acceleration,
and the parallel imaging acceleration factor in PE direction.

### Radiofrequency (RF) field mapping

> General notes:
> 
> -   Some RF file collections call for the use of special notations that cannot be resolved by
>     by entities that can generalize to other applications.
>     Instead of introducing an entity that is exclusive to a single application,
>     method developers who commonly use these file collections for the `MPM` application reached
>     the consensus on the use of `acq` entity to distinguish individual files.
>     These suffixes include: `TB1AFI`, `TB1TFL`, `TB1RMF`, and `RB1COR`.

Reviewer 3: This feels more like a prefatory paragraph than a bullet point.

```suggestion
Some RF file collections call for the use of special notations that cannot be resolved by
by entities that can generalize to other applications.
Instead of introducing an entity that is exclusive to a single application,
method developers who commonly use these file collections for the `MPM` application reached
the consensus on the use of `acq` entity to distinguish individual files.
These suffixes include: `TB1AFI`, `TB1TFL`, `TB1RMF`, and `RB1COR`.
```
